code/modules/mob/living/simple_animal/bot_swarm/swarmer.dm

```diff
@@ -0,0 +1,500 @@
+		return
+	Fabricate(/obj/effect/swarmer/trap, 5)
+	return
+
+/obj/effect/swarmer/trap
+	name = "swarmer trap"
+	desc = "A quickly assembled electric trap. Will not retain its form if damaged enough."
+	icon_state = "trap"
+	luminosity = 1
+	health = 10
+
+/obj/effect/swarmer/trap/Crossed(var/atom/movable/AM)
+	if(isliving(AM))
+		var/mob/living/L = AM
+		if(!istype(L, /mob/living/simple_animal/hostile/swarmer))
+			L.electrocute_act(0, src, 1, 1)
+			qdel(src)
+	..()
+
+/mob/living/simple_animal/hostile/swarmer/proc/CreateBarricade()
+	set name = "Create barricade"
+	set category = "Swarmer"
+	set desc = "Creates a barricade that will stop anything but swarmers and disabler beams from passing through."
+	if(/obj/effect/swarmer/blockade in loc)
+		src << "<span class='warning'>There is already a blockade here. Aborting.</span>"
+		return
+	Fabricate(/obj/effect/swarmer/blockade, 5)
+	return
+
+/obj/effect/swarmer/blockade
+	name = "swarmer blockade"
+	desc = "A quickly assembled energy blockade. Will not retain its form if damaged enough, but disabler beams and swarmers pass right through."
+	icon_state = "barricade"
+	luminosity = 1
+	health = 50
+	density = 1
+	anchored = 1
+
+/obj/effect/swarmer/blockade/CanPass(atom/movable/O)
+	if(isswarmer(O))
+		return 1
+	if(istype(O, /obj/item/projectile/beam/disabler))
+		return 1
+
+/mob/living/simple_animal/hostile/swarmer/proc/CreateSwarmer()
+	set name = "Replicate"
+	set category = "Swarmer"
+	set desc = "Creates a shell for a new swarmer. Swarmers will self activate."
+	src << "<span class='info'>We are attempting to replicate ourselves. We will need to stand still until the process is complete.</span>"
+	if(resources < 50)
+		src << "<span class='warning'>We do not have the resources for this!</span>"
+		return
+	if(!isturf(loc))
+		src << "<span class='warning'>This is not a suitable location for replicating ourselves. We need more room.</span>"
+		return
+	if(do_mob(src, src, 100))
+		if(Fabricate(/obj/item/unactivated_swarmer, 50))
+			playsound(loc,'sound/items/poster_being_created.ogg',50, 1, -1)
+
+/mob/living/simple_animal/hostile/swarmer/proc/RepairSelf()
+	set name = "Self Repair"
+	set category = "Swarmer"
+	set desc = "Attempts to repair damage to our body. You will have to remain motionless until repairs are complete."
+	if(!isturf(loc))
+		return
+	src << "<span class='info'>Attempting to repair damage to our body, stand by...</span>"
+	if(do_mob(src, src, 100))
+		adjustBruteLoss(-100)
+		src << "<span class='info'>We successfully repaired ourselves.</span>"
+
+/mob/living/simple_animal/hostile/swarmer/proc/ToggleLight()
+	if(!luminosity)
+		SetLuminosity(3)
+	else
+		SetLuminosity(0)
+
+/mob/living/simple_animal/hostile/swarmer/proc/ContactSwarmers()
+	var/message = input(src, "Announce to other swarmers", "Swarmer contact")
+	if(message)
+		for(var/mob/living/simple_animal/hostile/swarmer/S in mob_list)
```

Comment:
Should probably be visible to ghosts as well, so that you can ping for ghosts easier.